File: bar.js
module.exports = function bar(used, total, chars = ['#', '-'], max = 40) {

	var percent = used / total,
		len = Math.round(percent * max),
		rest = max - len

	percent = (percent * 100).toFixed(1)
	return ` ${percent}%${' '.repeat(6-percent.length)}[${chars[0].repeat(len)}\x1b[0m${chars[1].repeat(rest)}\x1b[0m]    `
}

File: bar.py
import math

def bar(used, total, chars=['#', '-'], max=40):
	percent = int(used)/float(total)
	len = int(math.floor(percent*max))
	rest = max - len

	return " {:.1f}% {}[{}\x1b[0m{}\x1b[0m]".format(percent*100, 'a' if percent>=1 else (' ' if percent >=0.1 else '  '), chars[0]*len, chars[1]*rest)
